app/presenters/curation_concerns_show_presenter.rb

```diff
@@ -5,6 +5,10 @@ def state_badge
     StateBadge.new(type, state).render
   end
 
+  def flaggable?
+    StateBadge.new(type, state).flaggable?
```

Comment:
Extract `StateBadge.new(type, state)` to a private method, and delegate flaggable? Also used in the above method.